doc: legacy withdrawal cell (#803)
nervosnetwork/godwoken: docs/deposit_and_withdrawal.md

### Legacy withdrawal cells (v0)

The lagacy withdrawal cells are not used anymore on the Godwoken network.

The withdrawal cell:

``` yaml
lock:
  code_hash:    (withdrawal lock's code hash),
  hash_type:    Type,
  args: (rollup_type_hash(32 bytes) | WithdrawalLockArgsV0 (n bytes) | owner lock len (optional) | owner lock (optional) | withdrawal_to_v1 flag byte (optional)
capacity:   (CKB amount),
type_:  (none or SUDT script)
data:   (none or SUDT amount)
```

Withdrawal lock guarantees the cell can only be unlocked after `finality blocks`.

```
// --- withdrawal lock ---
// a rollup_type_hash exists before this args, to make args friendly to prefix search
struct WithdrawalLockArgsV0 {
    account_script_hash: Byte32,
    withdrawal_block_hash: Byte32,
    withdrawal_block_number: Uint64,
    // buyer can pay sell_amount token to unlock
    sudt_script_hash: Byte32,
    sell_amount: Uint128,
    sell_capacity: Uint64,
    // layer1 lock to withdraw after challenge period
    owner_lock_hash: Byte32,
    // layer1 lock to receive the payment, must exists on the chain
    payment_lock_hash: Byte32,
}
```

> We have optional fields in the withdrawal cell's args:

Reviewer: ```suggestion
We have **optional** fields in the withdrawal cell's args:
```

> * owner lock - If users submit withdrawal request with a owner lock structure. The block producer will generate withdrawal cells with `owner lock` field in the args, and automatically unlock these cells after they'are finalized. Users are no need to manually unlock layer-1 withdrawal cells.

Reviewer: ```suggestion
* owner lock - If users submit withdrawal request with an owner lock structure. The block producer will generate withdrawal cells with `owner lock` field in the args, and automatically unlock these cells after they are finalized. Users don't need to manually unlock layer-1 withdrawal cells.
```

> * withdrawal_to_v1 - This field only works when `owner lock` exist, if `withdrawal_to_v1` is exist and the value is `1`, which means the withdrawal is a fast withdraw to v1 Godwoken, the withdraw can be instantly processed, and the assets will be migrated to v1 Godwoken.

Reviewer: ```suggestion
* withdrawal_to_v1 - This field only works when `owner lock` exist, if `withdrawal_to_v1` is exist and the value is `1`, which means the withdrawal is a fast withdrawal to Godwoken v1. A fast withdrawal from v0 to v1 can be instantly processed, and the assets will be migrated to Godwoken v1.
```

> * manually withdrawl - If `owner lock` do not exist, users must manually unlock the legacy withdrawal cell after it finalized, and user must provides an input cell in the unlocking transaction that it's `lock hash` is equals to withdrawal lock args' `owner_lock_hash`.

Reviewer: ```suggestion

### **Note**:
manually withdrawl - If `owner lock` do not exist, users must manually unlock the legacy withdrawal cell after it is finalized, and user must provides an input cell in the unlocking transaction that its `lock hash` is equals to withdrawal lock args' `owner_lock_hash`.
```

struct WithdrawalLockArgsV0 {
    account_script_hash: Byte32,
    withdrawal_block_hash: Byte32,
    withdrawal_block_number: Uint64,
    // buyer can pay sell_amount token to unlock
    sudt_script_hash: Byte32,
    sell_amount: Uint128,
    sell_capacity: Uint64,
    // layer1 lock to withdraw after challenge period
    owner_lock_hash: Byte32,
    // layer1 lock to receive the payment, must exists on the chain
    payment_lock_hash: Byte32,
}
```

We have **optional** fields in the withdrawal cell's args:

* owner lock - If users submit withdrawal request with an owner lock structure. The block producer will generate withdrawal cells with `owner lock` field in the args, and automatically unlock these cells after they are finalized. Users don't need to manually unlock layer-1 withdrawal cells.
* withdrawal_to_v1 - This field only works when `owner lock` exist, if `withdrawal_to_v1` is exist and the value is `1`, which means the withdrawal is a fast withdrawal to Godwoken v1. A fast withdrawal from v0 to v1 can be instantly processed, and the assets will be migrated to Godwoken v1.

### **Note**:

> manually withdrawl - If `owner lock` do not exist, users must manually unlock the legacy withdrawal cell after it is finalized, and user must provides an input cell in the unlocking transaction that its `lock hash` is equals to withdrawal lock args' `owner_lock_hash`.

Reviewer 2: do not exist -> does not exist
